### PR TITLE
Add generators and shrinkers for `SelectionLimit` and `SelectionSkeleton`.

### PR DESCRIPTION
**Pull request by jonathanknowles**
### Comments

This PR adds generators and shrinkers for `SelectionLimit` and `SelectionSkeleton`.

### Issue Number

ADP-1118

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Balance.hs [hunk 1]
```diff
@@ -233,12 +235,21 @@ emptySkeleton = SelectionSkeleton
...
+data SelectionLimitOf a
     = NoLimit
       -- ^ Indicates that there is no limit.
-    | MaximumInputLimit Int
+    | MaximumInputLimit a
       -- ^ Indicates a maximum limit on the number of inputs to select.
-    deriving (Eq, Show)
+    deriving (Eq, Functor, Show)
+
+instance Ord a => Ord (SelectionLimitOf a) where
```
Comment (sevanspowell) [inline lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Balance.hs:247]: Just to confirm, the Ord instance of SelectionLimitOf treats NoLimit is treated like "infinity"? If so, this instance makes sense.

**Review by sevanspowell — APPROVED** [review]

**Comment by sevanspowell** [comment]
Looks good to me :+1:

File: lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Balance.hs (hunk 1 above)
Comment (jonathanknowles) [reply lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Balance.hs:247]: Yes, that's correct!

For completeness (and to satisfy the `Ord` laws), we must also state that `NoLimit == NoLimit`.

**Comment by jonathanknowles** [comment]
bors retry

**Pull request merged**